crates/graphql_artifact_generation/src/generate_artifacts.rs

```diff
@@ -621,6 +622,25 @@ fn format_type_for_js_inner(
     }
 }
 
+pub(crate) fn generate_parameters<'a>(
+    schema: &ValidatedSchema,
+    argument_definitions: impl Iterator<Item = &'a VariableDefinition<SelectableServerFieldId>>,
+) -> String {
+    let mut s = "{\n".to_string();
+    let indent = "  ";
+    for arg in argument_definitions {
+        let is_optional = !matches!(arg.type_, GraphQLTypeAnnotation::NonNull(_));
+        s.push_str(&format!(
+            "{indent}readonly {}{}: {},\n",
+            arg.name.item,
+            if is_optional { "?" } else { "" },
```

Comment:
I wonder if we should make optional fields also `?`, not just optional arguments for consistency. Will DM you, this looks amazing and can land

Reply:
#160